pkg/apis/application/v1alpha1/types.go

```diff
@@ -141,6 +141,8 @@ type ApplicationSourceKustomize struct {
 	Images []string `json:"images,omitempty" protobuf:"bytes,3,opt,name=images"`
 	// CommonLabels adds additional kustomize commonLabels
 	CommonLabels map[string]string `json:"commonLabels,omitempty" protobuf:"bytes,4,opt,name=commonLabels"`
+	// BuildOptions is a string of parameters called with `kustomize build`
+	BuildOptions string `protobuf:"bytes,5,opt,name=buildOptions"`
```

Comment:
I'm questioning if build options should be at the app level because of any potential security concerns (e.g. --enable_alpha_plugins).

Reply:
Let's remove this on the safer side. It's impossible to predict what options will come in the future from Kustomize which would make this suddenly unsafe to execute as a per-app configuration. I worry `--enable_alpha_plugins` and combination of setting XDG_CONFIG_HOME would allow execution of arbitrary code in repo server

Reply:
What do you think of only allowing a predefined list of acceptable flags?

Reply:
I just don't see the use case for per-app build options at the moment, so rather keep the code simpler.

Reply:
Sounds good, will make the changes and ping you back